sql/2020/22_HTTP_2/avg_percentage_of_resources_loaded_over_HTTP_by_version_per_site.sql

```diff
@@ -0,0 +1,29 @@
+# standardSQL
+# Average percentage of resources loaded over HTTP .9, 1, 1.1, 2, and QUIC per site
+SELECT
+  client,
+  ROUND(AVG(http_0_9/num_requests) * 100,2) avg_pct_http_0_9,
+  ROUND(AVG(http_1_0/num_requests) * 100,2) avg_pct_http_1_0,  
+  ROUND(AVG(http_1_1/num_requests) * 100,2) avg_pct_http_1_1,
+  ROUND(AVG(http_2/num_requests) * 100,2) avg_pct_http_2,
+  ROUND(AVG(quic/num_requests) * 100,2) avg_pct_quic
+FROM (
+  SELECT 
+    client,
+    page,
+    COUNT(0) AS num_requests, 
+    SUM(IF(JSON_EXTRACT_SCALAR(payload, "$._protocol") ="http/0.9",1,0)) AS http_0_9, 
+    SUM(IF(JSON_EXTRACT_SCALAR(payload, "$._protocol") ="http/1.0",1,0)) AS http_1_0, 
+    SUM(IF(JSON_EXTRACT_SCALAR(payload, "$._protocol") ="http/1.1",1,0)) AS http_1_1,
+    SUM(IF(JSON_EXTRACT_SCALAR(payload, "$._protocol") ="HTTP/2",1,0)) AS http_2,
+    SUM(IF(JSON_EXTRACT_SCALAR(payload, "$._protocol") ="QUIC",1,0)) AS quic    
+  FROM 
+    `httparchive.almanac.requests`
+  WHERE
+    date = '2020-09-01'
+  GROUP BY
+    client,
+    page
+)
+GROUP BY 
+  client
```

Comment:
Minor reorganization to make the protocol variable and reshape the results, also adding an "other" bucket to account for everything else.

```suggestion
# standardSQL
# Average percentage of resources loaded over HTTP .9, 1, 1.1, 2, and QUIC per site
SELECT
  client,
  protocol,
  AVG(num_requests / total_requests) AS avg_pct_requests_per_page
FROM (
  SELECT 
    client,
    page,
    IF(JSON_EXTRACT_SCALAR(payload, '$._protocol') IN ('http/0.9', 'http/1.0', 'http/1.1', 'HTTP/2', 'QUIC'), JSON_EXTRACT_SCALAR(payload, '$._protocol'), 'other') AS protocol,
    COUNT(0) AS num_requests,
    SUM(COUNT(0)) OVER (PARTITION BY client, page) AS total_requests
  FROM 
    `httparchive.almanac.requests`
  WHERE
    date = '2020-09-01'
  GROUP BY
    client,
    page,
    protocol)
GROUP BY 
  client,
  protocol
ORDER BY
  avg_pct_requests_per_page DESC
```

Reply:
I've added the results for this new query to the sheet: https://docs.google.com/spreadsheets/d/1M1tijxf04wSN3KU0ZUunjPYCrVsaJfxzuRCXUeRQ-YU/edit#gid=119448240